QuoteGeneration/quote_wrapper/quote/id_enclave/id_enclave.cpp

```diff
@@ -288,3 +308,201 @@ sgx_status_t ide_get_pce_encrypt_key(
 
     return sgx_status;
 }
+
+sgx_status_t ide_get_ppid_encrypt_key(
+        const sgx_target_info_t* p_pce_target_info,
+        sgx_report_t* p_ide_report,
+        uint8_t crypto_suite,
+        uint16_t cert_key_type,
+        uint32_t key_size,
+        uint8_t* p_public_key)
+{
+    ref_rsa_params_t g_rsa_key = { 0 };  // The private key used to encrypt the PPID.  Only used for PPID_CEARTEXT Cert_Data_Type
+    return get_pce_encrypt_decrypt_key(p_pce_target_info, p_ide_report, crypto_suite, cert_key_type, key_size, p_public_key, &g_rsa_key);
+}
+
+sgx_status_t ide_decrypt_ppid(
+        const sgx_target_info_t* p_pce_target_info,
+        sgx_report_t* p_ide_report,
+        uint8_t crypto_suite,
+        uint16_t cert_key_type,
+        uint32_t key_size,
+        uint8_t* p_public_key,
+        uint32_t ppid_size,
+        uint8_t* ppid)
+{
+    ref_rsa_params_t g_rsa_key = { 0 };  // The private key used to encrypt the PPID.  Only used for PPID_CEARTEXT Cert_Data_Type
+    get_pce_encrypt_decrypt_key(p_pce_target_info, p_ide_report, crypto_suite, cert_key_type, key_size, p_public_key, &g_rsa_key);
```

Comment:
I recreate the decryption key here because it is created from a constant data. I do this because I think that moving a key from ` ide_get_ppid_encrypt_key` is complicated as it would require me to specify all the internal components in pointer format, so to not do this I decided to recreate the key twice. I accept better suggestions on how to do this.

Reply:
I don't get what you're trying to say. Let's back all the way up. We need to access the PCE enclave to get the PPID as the PCE enclave is the only one who can access it. Unfortunately, it doesn't allow us to just ask it for it. It requires a public key that it will use for encryption of the ppid. The idea now is to create an enclave (the id_enclave is very close to what you need, so you can start from there) that has one entry point: a `get_plaintext_ppid`. That entry point will first create an rsa key, then request the PCE enclave to get the PPID encrypted with the newly created public key, and then it will decrypt the ppid, and finally return it in plaintext.
The complication here is that the enclave needs to be able to contact the PCE enclave. There are two options:
 - The nicest way is to specify an untrusted function in the edl file to do that. That function will then become accessible from inside the enclave. For this to work, you can keep the generated RSA keys on the stack.
 - A less nice way would be to have two entry functions to the enclave. One to create all the required data (e.g., the public key) to send to the PCE enclave, and another to decrypt the encrypted ppid. In this option you can keep the generated RSA keys in a static variable within the enclave.